v2/reader_test.go

```diff
@@ -0,0 +1,246 @@
+package car_test
+
+import (
+	"io"
+	"os"
+	"testing"
+
+	carv2 "github.com/ipld/go-car/v2"
+	"github.com/ipld/go-car/v2/index"
+	"github.com/ipld/go-car/v2/internal/carv1"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReadVersion(t *testing.T) {
+	tests := []struct {
+		name    string
+		path    string
+		want    uint64
+		wantErr bool
+	}{
+		{
+			"CarV1VersionIsOne",
+			"testdata/sample-v1.car",
+			1,
+			false,
+		},
+		{
+			"CarV2VersionIsTwo",
+			"testdata/sample-rw-bs-v2.car",
+			2,
+			false,
+		},
+		{
+			"CarV1VersionWithZeroLenSectionIsOne",
+			"testdata/sample-v1-with-zero-len-section.car",
+			1,
+			false,
+		},
+		{
+			"AnotherCarV1VersionWithZeroLenSectionIsOne",
+			"testdata/sample-v1-with-zero-len-section2.car",
+			1,
+			false,
+		},
+		{
+			"WrappedCarV1InCarV2VersionIsTwo",
+			"testdata/sample-wrapped-v2.car",
+			2,
+			false,
+		},
+		{
+			"FutureVersionWithCorrectPragmaIsAsExpected",
+			"testdata/sample-rootless-v42.car",
+			42,
+			false,
+		},
+		{
+			"CarV1WithValidHeaderButCorruptSectionIsOne",
+			"testdata/sample-v1-tailing-corrupt-section.car",
+			1,
+			false,
+		},
+		{
+			"CarV2WithValidHeaderButCorruptSectionAndIndexIsTwo",
+			"testdata/sample-v2-corrupt-data-and-index.car",
+			2,
+			false,
+		},
+		{
+			"CarFileWithCorruptPragmaIsError",
+			"testdata/sample-corrupt-pragma.car",
+			0,
+			true,
```

Comment:
please key all these fields :) as a bonus, you can then get rid of the lines where the values are zero (0, false, etc)